addons/onboarding/models/onboarding_progress.py

```diff
@@ -33,15 +35,21 @@ def init(self):
 
     @api.depends('onboarding_id.step_ids', 'progress_step_ids', 'progress_step_ids.step_state')
     def _compute_onboarding_state(self):
-        progress_steps_data = self.env['onboarding.progress.step'].read_group(
-            [('progress_id', 'in', self.ids), ('step_state', 'in', ['just_done', 'done'])],
-            ['progress_id'], ['progress_id']
-        )
-        result = dict((data['progress_id'][0], data['progress_id_count']) for data in progress_steps_data)
         for progress in self:
             progress.onboarding_state = (
-                'not_done' if result.get(progress.id, 0) != len(progress.onboarding_id.step_ids)
-                else 'done')
+                'not_done' if (
+                    len(progress.progress_step_ids.filtered(lambda p: p.step_state in {'just_done', 'done'}))
+                    != len(progress.onboarding_id.step_ids)
+                )
+                else 'done'
+            )
+
+    @api.depends('onboarding_id.step_ids')
+    def _compute_progress_step_ids(self):
+        """Updates progress steps when a step (with existing progress) is added
+        to an onboarding."""
+        for progress in self:
+            progress.progress_step_ids = progress.onboarding_id.step_ids.current_progress_step_id.filtered(bool)
```

Comment:
I think the word is not "clever" but "useless" in this case :D (It's a remnant of another ineffective implementation where I filtered `id`s and had to discard `False`).

(I also thought newIds were "existing" though, which is unpractical in some places, btw)

Reply:
The only thing it can do is remove the `NewId` records, but I really doubt that it is intended.
The getter on relational fields already takes only the records that exist.